rotkehlchen/tests/unit/accounting/test_default_settings.py

```diff
@@ -0,0 +1,332 @@
+        rate=Price(FVal(1862.06)),
+        index=0,
+    )
+    acquisition_event.remaining_amount = FVal(0.5)
+    cost_basis = None
+    if is_taxable:
+        cost_basis = CostBasisInfo(
+            taxable_amount=FVal(0.5) if is_taxable else ZERO,
+            taxable_bought_cost=FVal(931.03),
+            taxfree_bought_cost=ZERO,
+            matched_acquisitions=[
+                MatchedAcquisition(
+                    amount=FVal(0.5),
+                    event=acquisition_event,
+                    taxable=True,
+                ),
+            ],
+            is_complete=True,
+        )
+    expected_event = ProcessedAccountingEvent(
+        type=AccountingEventType.TRANSACTION_EVENT,
+        notes='Send 1 ETH to 0xABC',
+        location=Location.BLOCKCHAIN,
+        timestamp=TIMESTAMP_2_SECS,
+        asset=A_ETH,
+        free_amount=ZERO,
+        taxable_amount=FVal(0.5),
+        price=Price(FVal(3258.55)),
+        pnl=PNL(
+            taxable=FVal(-931.03) if is_taxable else ZERO,
+            free=ZERO,
+        ),
+        cost_basis=cost_basis,
+        index=1,
+        extra_data={'tx_hash': EXAMPLE_TX_HASH_HEX},
+    )
+    expected_event.count_entire_amount_spend = is_taxable
+    expected_event.count_cost_basis_pnl = is_taxable
+
+    assert accounting_pot.processed_events[-1] == expected_event
+    assert accounting_pot.pnls.taxable == FVal(1862.06) + expected_event.pnl.taxable
+    assert accounting_pot.pnls.free == ZERO
+
+
+@pytest.mark.parametrize('mocked_price_queries', [prices])
+def test_accounting_swap_settings(accounting_pot: 'AccountingPot'):
+    """Test that the default accounting settings for swaps are correct"""
+    _gain_one_ether(transactions=accounting_pot.transactions)
+
+    swap_spend_event = HistoryBaseEntry(
+        event_identifier=EXAMPLE_EVM_HASH,
+        sequence_index=1,
+        timestamp=TIMESTAMP_2_MS,
+        location=Location.BLOCKCHAIN,
+        location_label=EXAMPLE_ADDRESS,
+        asset=A_ETH,
+        balance=Balance(amount=ONE),
+        notes='Swap 1 ETH in a uniswap pool',
+        event_type=HistoryEventType.TRADE,
+        event_subtype=HistoryEventSubType.SPEND,
+    )
+    swap_receive_event = HistoryBaseEntry(
+        event_identifier=EXAMPLE_EVM_HASH,
+        sequence_index=2,
+        timestamp=TIMESTAMP_2_MS,
+        location=Location.BLOCKCHAIN,
+        location_label=EXAMPLE_ADDRESS,
+        asset=A_DAI,
+        balance=Balance(amount=FVal(1200)),
+        notes='Receive 1200 DAI as the result of a swap',
+        event_type=HistoryEventType.TRADE,
+        event_subtype=HistoryEventSubType.RECEIVE,
+    )
+    consumed_num = accounting_pot.transactions.process(
+        event=swap_spend_event,
+        events_iterator=iter([swap_receive_event]),
+    )
+    assert consumed_num == 2
+    acquisition_event = AssetAcquisitionEvent(
+        amount=ONE,
+        timestamp=TIMESTAMP_1_SECS,
+        rate=Price(FVal(1862.06)),
```

Comment:
same comments for rates here. Also for numbers you use multiple times in a test, add them in a variable